tools/oeutil/host/generate_evidence.cpp

```diff
@@ -29,18 +35,72 @@
 
 #include "../../../common/attest_plugin.h"
 #include "../../../common/sgx/collateral.h"
+#include "../../../common/sgx/endorsements.h"
 #include "../../../common/sgx/quote.h"
 #include "../../../host/sgx/sgxquoteprovider.h"
-
+#include "generate_evidence.h"
+#include "parse_args_helper.h"
 extern FILE* log_file;
 
-#define OE_PEM_BEGIN_CERTIFICATE "-----BEGIN CERTIFICATE-----"
-#define OE_PEM_BEGIN_CERTIFICATE_LEN (sizeof(OE_PEM_BEGIN_CERTIFICATE) - 1)
-#define OE_PEM_END_CERTIFICATE "-----END CERTIFICATE-----"
-#define OE_PEM_END_CERTIFICATE_LEN (sizeof(OE_PEM_END_CERTIFICATE) - 1)
+#define DEFAULT_LOG_FILE "oeutil_generate_evidence.log"
+#define ENCLAVE_FILENAME_SUFFIX "_enc.signed"
+#define INPUT_PARAM_OPTION_FORMAT "--format"
+#define INPUT_PARAM_OPTION_CERT "cert"
+#define INPUT_PARAM_OPTION_ENDORSEMENTS_FILENAME "--endorsements"
+#define INPUT_PARAM_OPTION_QUOTE_PROC "--quote-proc"
+#define INPUT_PARAM_QUOTE_IN_PROC "in"
+#define INPUT_PARAM_QUOTE_OUT_OF_PROC "out"
+#define INPUT_PARAM_OPTION_VERIFY "--verify"
+#define INPUT_PARAM_OPTION_OUT_FILE "--out"
+#define INPUT_PARAM_OPTION_LOG_FILE "--log"
+#define INPUT_PARAM_OPTION_VERBOSE "--verbose"
+#define INPUT_PARAM_OPTION_HELP "--help"
+#define INPUT_PARAM_OPTION_LEGACY_REPORT_REMOTE "LEGACY_REPORT_REMOTE"
+#define INPUT_PARAM_OPTION_SGX_ECDSA "SGX_ECDSA"
+#define INPUT_PARAM_OPTION_SGX_EPID_LINKABLE "SGX_EPID_LINKABLE"
+#define INPUT_PARAM_OPTION_SGX_EPID_UNLINKABLE "SGX_EPID_UNLINKABLE"
+#define SHORT_INPUT_PARAM_OPTION_FORMAT "-f"
+#define SHORT_INPUT_PARAM_OPTION_ENDORSEMENTS_FILENAME "-e"
+#define SHORT_INPUT_PARAM_OPTION_QUOTE_PROC "-p"
+#define SHORT_INPUT_PARAM_OPTION_VERIFY "-v"
+#define SHORT_INPUT_PARAM_OPTION_OUT_FILE "-o"
+#define SHORT_INPUT_PARAM_OPTION_LOG_FILE "-l"
+#define SHORT_INPUT_PARAM_OPTION_HELP "-h"
+#define SGX_AESM_ADDR "SGX_AESM_ADDR"
+#if defined(_WIN32)
+#define SGX_AESM_ADDR_MAXSIZE 32
+#endif
+
 #define SGX_EXTENSION_OID_STR "1.2.840.113741.1.13.1"
 #define MAX_BUFFER_SIZE 65536
 
+// Static constants for evidence UUIDs
+static const oe_uuid_t _sgx_ecdsa_uuid = {OE_FORMAT_UUID_SGX_ECDSA};
+static const oe_uuid_t _sgx_epid_linkable_uuid = {
+    OE_FORMAT_UUID_SGX_EPID_LINKABLE};
+static const oe_uuid_t _sgx_epid_unlinkable_uuid = {
+    OE_FORMAT_UUID_SGX_EPID_UNLINKABLE};
+
+// Structure to store input parameters
+typedef struct _input_params
+{
+    const char* private_key_filename;
+    const char* public_key_filename;
+    const char* out_filename;
+    const char* log_filename;
+    const char* endorsements_filename;
+    const char* quote_proc;
+    bool generate_certificate;
+    bool generate_legacy_report_remote;
+    bool generate_sgx_ecdsa;
+    bool generate_sgx_epid_linkable;
+    bool generate_sgx_epid_unlinkable;
+    bool verify;
+    bool verbose;
+} input_params_t;
```

Comment:
```suggestion
} input_parameters_t;
```
avoid abbreviations per OE style guidance (also in line 85)

Reply:
Sorry missed those. Updated.